termuic/termuiConsole.go

```diff
@@ -0,0 +1,276 @@
+
+	grid := tc.configConsoleWithNormalTermuiWidgets()
+
+	uiEvents := ui.PollEvents()
+	// handles kill signal sent to gotop
+	sigTerm := make(chan os.Signal, 2)
+	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
+
+	drawTicker := time.NewTicker(time.Second).C
+
+	for {
+		select {
+		case <-drawTicker:
+			tc.refreshDataForConsole()
+
+			ui.Clear()
+			ui.Render(grid)
+		case <-sigTerm:
+			ui.Clear()
+
+			return
+		case e := <-uiEvents:
+			switch e.ID {
+			case "<Resize>":
+				payload := e.Payload.(ui.Resize)
+				grid.SetRect(0, 0, payload.Width, payload.Height)
+				ui.Clear()
+				ui.Render(grid)
+			case "q":
+				ui.Clear()
+				grid = tc.changeConsoleDisplay()
+
+			case "<C-c>":
+				ui.Clear()
+				_ = syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+
+				return
+			}
+		}
+	}
+}
+
+func (tc *TermuiConsole) refreshDataForConsole() {
+	if tc.viewBigLog == false {
+		tc.prepareDataNormalWidgets()
+	} else {
+		tc.prepareDataBigLog()
+	}
+}
+
+func (tc *TermuiConsole) prepareDataNormalWidgets() {
+	nonceI, _ := tc.TermuiConsoleMetrics.Load(core.MetricNonce)
+	nonce := nonceI.(int)
+
+	synchronizedRoundI, _ := tc.TermuiConsoleMetrics.Load(core.MetricSynchronizedRound)
+	synchronizedRound := synchronizedRoundI.(int)
+
+	currentRoundI, _ := tc.TermuiConsoleMetrics.Load(core.MetricCurrentRound)
+	currentRound := currentRoundI.(int)
+
+	isSyncingI, _ := tc.TermuiConsoleMetrics.Load(core.MetricIsSyncing)
+	isSyncing := isSyncingI.(int)
+
+	tc.termuiConsoleWidgets.PrepareSyncInfoForDisplay(nonce, currentRound, synchronizedRound, isSyncing)
+
+	publicKeyI, _ := tc.TermuiConsoleMetrics.Load(core.MetricPublicKey)
+	publicKey := publicKeyI.(string)
+	tc.termuiConsoleWidgets.PreparePublicKeyForDisplay(publicKey)
+
+	shardIdI, _ := tc.TermuiConsoleMetrics.Load(core.MetricShardId)
+	shardId := shardIdI.(int)
+	tc.termuiConsoleWidgets.PrepareShardIdForDisplay(shardId)
+
+	numConnectedPeersI, _ := tc.TermuiConsoleMetrics.Load(core.MetricNumConnectedPeers)
+	numConnectedPeers := numConnectedPeersI.(int)
+
+	txPoolLoadI, _ := tc.TermuiConsoleMetrics.Load(core.MetricTxPoolLoad)
+	txPoolLoad := txPoolLoadI.(int)
+	tc.termuiConsoleWidgets.PrepareTxPoolLoadForDisplay(txPoolLoad)
+	tc.termuiConsoleWidgets.PrepareNumConnectedPeersForDisplay(numConnectedPeers)
+
+	countConsensusI, _ := tc.TermuiConsoleMetrics.Load(core.MetricCountConsensus)
+	countConsensus := countConsensusI.(int)
+
+	countLeaderI, _ := tc.TermuiConsoleMetrics.Load(core.MetricCountLeader)
+	countLeader := countLeaderI.(int)
+
+	countAcceptedBlocksI, _ := tc.TermuiConsoleMetrics.Load(core.MetricCountAcceptedBlocks)
+	countAcceptedBlocks := countAcceptedBlocksI.(int)
+
+	tc.termuiConsoleWidgets.PrepareConcensusInformationsForDisplay(countConsensus, countLeader, countAcceptedBlocks)
+
+	tc.termuiConsoleWidgets.PrepareListWithLogsForDisplay(tc.logLines)
+}
+
+func (tc *TermuiConsole) prepareDataBigLog() {
+	tc.termuiConsoleWidgets.PrepareListWithLogsForDisplay(tc.logLines)
```

Comment:
a function that was created just to call other function?

Reply:
remove this function and where this function was called use direct this call
 " tc.termuiConsoleWidgets.PrepareListWithLogsForDisplay(tc.logLines) "